django/core/management/color.py

```diff
@@ -14,11 +14,57 @@ def supports_color():
     Return True if the running system's terminal supports color,
     and False otherwise.
     """
-    supported_platform = sys.platform != 'win32' or 'ANSICON' in os.environ
+
+    def supported_platform():
```

Comment:
Do we really need this inner function? We could just shortcut out early if not a TTY.